src/main/common/filter.c

```diff
@@ -122,16 +128,25 @@ void biquadFilterInit(biquadFilter_t *filter, uint16_t filterFreq, uint32_t refr
     filter->a1 = a1 / a0;
     filter->a2 = a2 / a0;
 
-    // zero initial samples
-    filter->d1 = filter->d2 = 0;
+    filter->filterFreq = filterFreq;
+    filter->refreshRate = refreshRate;
+    filter->running = false;
 }
 
 // Computes a biquad_t filter on a sample
 float biquadFilterApply(biquadFilter_t *filter, float input)
 {
-    const float result = filter->b0 * input + filter->d1;
-    filter->d1 = filter->b1 * input - filter->a1 * result + filter->d2;
-    filter->d2 = filter->b2 * input - filter->a2 * result;
+    float result;
+    if (!filter->running) {
```

Comment:
The biquad filter is a time-critical component and as such it is extremely undesirable that for it to contain any unnecessary code. The cost of the if statement should be borne by the code that requires it and should not be imposed on all users of the biquad filter.

So instead of adding a condition to the filter, you should have a separate condition at the point biquadFilterApply is called. (If it is called via function pointer at that point, you could handle this case by appropriate setting up of the function pointer.)